### PR TITLE
Fix compilation against latest Infinispan master

### PR DESCRIPTION
**Pull request by ryanemerson**

### CODE REVIEW

File: server/deployment/src/main/java/org/infinispan/quarkus/server/deployment/InfinispanServerProcessor.java [hunk 1]
```diff
@@ -206,6 +205,11 @@ void addReflectionAndResources(BuildProducer<ReflectiveClassBuildItem> reflectio
       reflectionClass.produce(new ReflectiveClassBuildItem(false, false, ClusteredLockManagerConfigurationBuilder.class));
    }
 
+   @BuildStep
+   void excludeResourcesFromDependencies(BuildProducer<ExcludeConfigBuildItem> excludeConfigBuildItemBuildProducer) {
+      excludeConfigBuildItemBuildProducer.produce(new ExcludeConfigBuildItem("io.lettuce.lettuce-core-6.1.6.RELEASE.jar", "/META-INF/native-image/io.lettuce/lettuce-core/reflect-config.json"));
```
Comment (wburns) [inline server/deployment/src/main/java/org/infinispan/quarkus/server/deployment/InfinispanServerProcessor.java:210]: I would say to use  regex for the version https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ExcludeConfigBuildItem.java#L9

**Review by wburns — COMMENTED** [review]
LGTM.

File: server/deployment/src/main/java/org/infinispan/quarkus/server/deployment/InfinispanServerProcessor.java (hunk 1 above)
Comment (ryanemerson) [reply server/deployment/src/main/java/org/infinispan/quarkus/server/deployment/InfinispanServerProcessor.java]: Done

**Comment by ryanemerson** [comment]
It seems the CI doesn't like the new `Micrometer` integration either

File: server/deployment/src/main/java/org/infinispan/quarkus/server/deployment/InfinispanServerProcessor.java [hunk 2]
```diff
@@ -206,6 +205,11 @@ void addReflectionAndResources(BuildProducer<ReflectiveClassBuildItem> reflectio
       reflectionClass.produce(new ReflectiveClassBuildItem(false, false, ClusteredLockManagerConfigurationBuilder.class));
    }
 
+   @BuildStep
+   void excludeResourcesFromDependencies(BuildProducer<ExcludeConfigBuildItem> excludeConfigBuildItemBuildProducer) {
+      excludeConfigBuildItemBuildProducer.produce(new ExcludeConfigBuildItem("io.lettuce.lettuce-core-*.+.jar", "/META-INF/native-image/io.lettuce/lettuce-core/reflect-config.json"));
```
Comment (wburns) [inline server/deployment/src/main/java/org/infinispan/quarkus/server/deployment/InfinispanServerProcessor.java:210]:
```suggestion
      excludeConfigBuildItemBuildProducer.produce(new ExcludeConfigBuildItem("io.lettuce.lettuce-core-.+.jar", "/META-INF/native-image/io.lettuce/lettuce-core/reflect-config.json"));
```

I wonder if we should escape the dots also...? But if it works then it should be fine.

**Comment by ryanemerson** [comment]
> We probably want to add micrometer to the classpath for the server. This should fix that error and if there is any native stuff needed Quarkus should already provide it.

It seems more is need. I added the dependencies and the error is the same. The `infinispan-server-rest` module already pulls in both dependencies, so I don't think explicitly defining them should be required :thinking:

**Review by wburns — APPROVED** [review]

**Pull request merged**